smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/endpoints/EndpointParamsGenerator.kt

```diff
@@ -9,28 +9,44 @@ import software.amazon.smithy.codegen.core.Symbol
 import software.amazon.smithy.rulesengine.language.EndpointRuleSet
 import software.amazon.smithy.rulesengine.language.syntax.parameters.Parameter
 import software.amazon.smithy.rulesengine.language.syntax.parameters.ParameterType
+import software.amazon.smithy.rulesengine.traits.EndpointRuleSetTrait
 import software.amazon.smithy.swift.codegen.AuthSchemeResolverGenerator
 import software.amazon.smithy.swift.codegen.SwiftTypes
 import software.amazon.smithy.swift.codegen.SwiftWriter
 import software.amazon.smithy.swift.codegen.integration.ProtocolGenerator
 import software.amazon.smithy.swift.codegen.model.boxed
 import software.amazon.smithy.swift.codegen.model.defaultValue
+import software.amazon.smithy.swift.codegen.model.getTrait
+import software.amazon.smithy.swift.codegen.swiftmodules.ClientRuntimeTypes
 import software.amazon.smithy.swift.codegen.utils.clientName
 import software.amazon.smithy.swift.codegen.utils.toLowerCamelCase
 
 /**
  * Generates EndpointParams struct for the service
  */
-class EndpointParamsGenerator(private val endpointRules: EndpointRuleSet?) {
-    fun render(writer: SwiftWriter, ctx: ProtocolGenerator.GenerationContext? = null) {
+class EndpointParamsGenerator(
+    val ctx: ProtocolGenerator.GenerationContext,
+) {
+    fun render() {
```

Comment:
Endpoint params now get written to their own source file rather than being lumped in with the endpoint resolver.